lib/ansible/modules/cloud/amazon/aws_organization_units.py

```diff
@@ -0,0 +1,272 @@
+            parent_ou_id = parent_ou["Id"]
+        return parent_ou
+
+    def get_aws_organizational_unit_by_arn(self, arn):
+        ou_id = arn.split("/")[-1]
+        try:
+            parent_ou = self.client.describe_organizational_unit(OrganizationalUnitId=ou_id)
+        except ClientError as e:
+            if "OrganizationalUnitNotFoundException" in e.response["Error"]["Message"]:
+                return None
+            else:
+                raise e
+        return parent_ou["OrganizationalUnit"]
+
+    def has_children(self):
+        if self.ou is None:
+            return False
+        paginator = self.client.get_paginator("list_children")
+        for child_type in ["ACCOUNT", "ORGANIZATIONAL_UNIT"]:
+            for page in paginator.paginate(ParentId=self.ou["Id"], ChildType=child_type):
+                if len(page["Children"]) > 0:
+                    return True
+        return False
+
+    def delete_aws_organizational_unit(self):
+        if self.ou is None:
+            return True
+        if self.has_children():
+            raise OrganizationalUnitDependencyException
+        self.client.delete_organizational_unit(OrganizationalUnitId=self.ou["Id"])
+        return True
+
+    def create_aws_organizational_unit(self):
+        if self.ou is not None:
+            return True
+        if self.name is None:
+            raise ParameterValidationException
+        parent_name = self.name.rsplit("/", 1)[0]
+        ou_name = self.name.split("/")[-1]
+        if parent_name != ou_name:
+            parent_ou = self.get_aws_organizational_unit_by_name(parent_name)
+            if parent_ou is None:
+                raise ParentOrganizationalUnitDoesNotExist
+            parent_id = parent_ou["Id"]
+        else:
+            parent_id = self.root["Id"]
+        self.ou = self.client.create_organizational_unit(ParentId=parent_id, Name=ou_name)["OrganizationalUnit"]
+        return self.ou
+
+
+def main():
+    argument_spec = dict(
+        name=dict(type='str'),
+        arn=dict(type='str'),
+        state=dict(default='present', choices=['absent', 'present']),
+    )
+
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        mutually_exclusive=[['name', 'arn']],
+        required_one_of=[['name', 'arn']],
+        required_if=[
+            ['state', 'present', ['name']]
+        ]
+    )
+
+    result = dict(
+        name=module.params.get('name'),
+        ou=dict(),
+        changed=False,
+        state='absent',
+    )
+
+    ou_name = module.params.get('name')
+    ou_arn = module.params.get('arn')
+    ou_state = module.params.get('state')
+
+    if ou_arn:
+        if not ou_arn.startswith("arn:aws:organizations::"):
+            module.fail_json(msg="Invalid ARN format")
+        try:
+            client = AwsOrganizationalUnit(module, arn=ou_arn)
+        except (BotoCoreError, ClientError) as e:
+            module.fail_json(msg="Boto failure")
```

Comment:
```suggestion
            module.fail_json_aws(e, msg="Boto failure")
```
Boto/API exceptions should use fail_json_aws